### ISSUE TITLE
Keyset pagination is not supported

### ISSUE THREAD

**Issue by max-wittig** [jdalrymple/gitbeaker#1115]
Keyset pagination is required, when you are requesting more than `50.000` projects.

When using keyset pagination with `gitbeaker`, you'll only get the first page back: 

```js
gitlab.Projects.all({'per_page': 100, 'statistics': true, 'pagination': 'keyset', 'order_by': 'id'})
```

https://docs.gitlab.com/ee/api/#pagination

**Expected behaviour**

gitbeaker should return more than 100 projects, when using keyset pagination

**Actual behaviour**

gitbeaker returns only 100 projects, when using keyset pagination

**Possible fixes**

Not sure. Something with the headers changed at GitLab and `gitbeaker` most likely is reading the wrong headers

**Comment by max-wittig** [comment]
`gitbeaker` actually parses the `X-Next` headers, which GitLab didn't recommend and now they removes them in keyset pagination. These are not coming back either. The link header needs to be parsed: https://github.com/jdalrymple/gitbeaker/blob/master/packages/gitbeaker-core/src/infrastructure/RequestHelper.ts#L77

**Comment by jdalrymple** [comment]
Unless im missing something in the docs, they still say its available https://docs.gitlab.com/ee/api/#pagination. Or are the docs just outdated?

**Comment by max-wittig** [comment]
I think GitLab is never gonna support all endpoints, but it **requires** it for the /projects endpoint, if you have more than 50.000 projects. 

You can't make it a default, but you need to get away from this unsupported way to paginate anyway, otherwise it's not gonna work at all. 

Using the X-Next-Page header was never supported and is dropped with `keyset pagination`. You need to parse the `Link` header instead and get next from that. That's what https://github.com/jdalrymple/gitbeaker/pull/1116/files fixes.

**Comment by max-wittig** [comment]
It's still kinda a bug 😄

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`24.0.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/24.0.0) :rocket:

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`24.0.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/24.0.0) :rocket:
